src/DataStructures/Python/ToNumpyArray.cpp

```diff
@@ -0,0 +1,31 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+#include "DataStructures/Python/ToNumpyArray.hpp"
+
+#include <array>
+#include <cstddef>
+#include <utility>
+#include "DataStructures/Python/Numpy.hpp"
+
+#include "DataStructures/Matrix.hpp"
+
+namespace py_bindings {
+
+PyObject* to_numpy_array(const Matrix& matrix) {
+  double* c_style_data = static_cast<double*>(
+      malloc(sizeof(double) * (matrix.rows()) * matrix.columns()));
+  for (size_t i = 0; i < matrix.rows(); ++i) {
+    for (size_t j = 0; j < matrix.columns(); ++j) {
+      c_style_data[j + i * matrix.columns()] = matrix(i, j);
+    }
+  }
+  std::array<long, 2> dims{
+      {static_cast<long>(matrix.rows()), static_cast<long>(matrix.columns())}};
+  PyObject* npy_array =
+      PyArray_SimpleNewFromData(2, dims.data(), NPY_DOUBLE, c_style_data);
+  PyArrayObject* npy_array_obj = reinterpret_cast<PyArrayObject*>(npy_array);
```

Comment:
Could you add a comment along the lines of:
```cpp
// The `reinterpret_cast` is intentional because we know the pointer actually points
// to an object of type `PyArrayObject` and we need to access it in that manner.
```
The reason is that `reinterpret_cast` is one of the very dark corners of C++ and so it's good practice to state that you are intentionally using that feature and why.